pkg/reconciler/reconciler.go

```diff
@@ -125,3 +125,15 @@ func init() {
 	// logged for pipeline types.
 	pipelineScheme.AddToScheme(scheme.Scheme)
 }
+
+// EmitEvent emits success or failed event for obj if condition has changed
```

Comment:
can you explain in this docstring a bit more about what the parameters do? i dont understand how "if condition has changed" relates to `beforeCondition` and `afterCondition`